### PR TITLE
libbraiding version bump

### LINKED ISSUE #38886: Upgrade to libbraiding v1.3
### Problem Description

Per request, a new version of libbraiding has been released, with new functions exposed.

### Proposed Solution

Upgrade to the new version and expose the new functions.

### Alternatives Considered

I attach a PR.

### Additional Information

_No response_

### Is there an existing issue for this?

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.

### PR DESCRIPTION
**Pull request by miguelmarco**
Fixes #38886 by updating libbraiding package and adding new methods to braids.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: src/sage/groups/braid.py [hunk 1]
```diff
@@ -2239,6 +2240,95 @@ def colored_jones_polynomial(self, N, variab=None, try_inverse=True):
         self._cj_with_q[N] = cj.subs({q: 1/q}) if use_inverse else cj
         return self.colored_jones_polynomial(N, variab, try_inverse)
 
+    def send_to_sss(self):
+        r"""
+        Return an element of the braid's super summit set, an the conjugating
```
Comment (tscrim) [inline src/sage/groups/braid.py]:
```suggestion
        Return an element of the braid's super summit set and the conjugating
```

File: src/sage/groups/braid.py [hunk 2]
```diff
@@ -2239,6 +2240,95 @@ def colored_jones_polynomial(self, N, variab=None, try_inverse=True):
...
+             s0*s1*s0*s2*s1]
+
+        """
+        traj = trajectory(self)
+        B = self.parent()
+        return [B._element_from_libbraiding(b) for b in traj]
+
+    def cyclic_slidings(self):
+        r"""
+        Return the the braid's cyclic slidings.
```
Comment (tscrim) [inline src/sage/groups/braid.py]:
```suggestion
        Return the braid's cyclic slidings.
```

File: src/sage/groups/braid.py [hunk 3]
```diff
@@ -2239,6 +2240,95 @@ def colored_jones_polynomial(self, N, variab=None, try_inverse=True):
         self._cj_with_q[N] = cj.subs({q: 1/q}) if use_inverse else cj
         return self.colored_jones_polynomial(N, variab, try_inverse)
 
+    def send_to_sss(self):
```
Comment (tscrim) [inline src/sage/groups/braid.py]: This method name is not very clear. How about `super_summit_set[_element]`?

File: src/sage/groups/braid.py [hunk 4]
```diff
@@ -2239,6 +2240,95 @@ def colored_jones_polynomial(self, N, variab=None, try_inverse=True):
...
+            sage: b = B([1, 2, 1, 2, 3, -1, 2, 1, 3])
+            sage: b.send_to_sss()
+            (s0*s2*s0*s1*s2*s1*s0, s0^-1*s1^-1*s0^-1*s2^-1*s1^-1*s0^-1*s1*s0*s2*s1*s0)
+
+        """
+        to_sss = send_to_sss(self)
+        B = self.parent()
+        return tuple(B._element_from_libbraiding(b) for b in to_sss)
+
+    def send_to_uss(self):
```
Comment (tscrim) [inline src/sage/groups/braid.py]: Same comment about the name.

File: src/sage/groups/braid.py [hunk 5]
```diff
@@ -2239,6 +2240,95 @@ def colored_jones_polynomial(self, N, variab=None, try_inverse=True):
...
+        r"""
+        Return an element of the braid's ultra summit set, an the conjugating
```
Comment (tscrim) [inline src/sage/groups/braid.py]:
```suggestion
        Return an element of the braid's ultra summit set and the conjugating
```

File: src/sage/groups/braid.py [hunk 6]
```diff
@@ -2239,6 +2240,95 @@ def colored_jones_polynomial(self, N, variab=None, try_inverse=True):
...
+        braid.
+
+        EXAMPLES::
+
+            sage: B = BraidGroup(4)
+            sage: b = B([1, 2, 1, 2, 3, -1, 2, 1, 3])
+            sage: b.send_to_sss()
+            (s0*s2*s0*s1*s2*s1*s0, s0^-1*s1^-1*s0^-1*s2^-1*s1^-1*s0^-1*s1*s0*s2*s1*s0)
+
+        """
```
Comment (tscrim) [inline src/sage/groups/braid.py]:
```suggestion
            (s0*s2*s0*s1*s2*s1*s0, s0^-1*s1^-1*s0^-1*s2^-1*s1^-1*s0^-1*s1*s0*s2*s1*s0)
        """
```
We generally do not have this blank line IIRC.

File: src/sage/libs/braiding.pyx [hunk 7]
```diff
@@ -383,3 +388,156 @@ def sliding_circuits(braid):
...
+    return rop
+
+
+def cyclic_slidings(braid):
+    r"""
+    Return the cyclic slidings of the braid.
+
+    INPUT:
+
+    - ``braid```-- a braid
```
Comment (tscrim) [inline src/sage/libs/braiding.pyx]:
```suggestion
    - ``braid``` -- a braid
```

File: src/sage/groups/braid.py [hunk 8]
```diff
@@ -2239,6 +2240,95 @@ def colored_jones_polynomial(self, N, variab=None, try_inverse=True):
...
+
+            sage: B = BraidGroup(4)
+            sage: b = B([1, 2, 1, 2, 3, -1, 2, -1, 3])
+            sage: b.send_to_uss()
+            (s0*s1*s0*s2*s1, s0^-1*s1^-1*s0^-1*s2^-1*s1^-1*s0^-1*s1*s2*s1^2*s0)
+
+        """
+        to_uss = send_to_uss(self)
+        B = self.parent()
+        return tuple(B._element_from_libbraiding(b) for b in to_uss)
```
Comment (tscrim) [inline src/sage/groups/braid.py]:
```suggestion
        return tuple([B._element_from_libbraiding(b) for b in to_uss])
```
For whatever reason, this is marginally faster.

**Review by tscrim — CHANGES_REQUESTED** [review]
A few little details. Otherwise LGTM.

**Comment by miguelmarco** [comment]
Thanks for the quick review!

There is still a problem with the install of the new version of libbraiding that I don't know how to solve: in the CI environments the new version is not installed, and then the tests fail.

I guess it has something to do with the `environment-3*.yaml` files, which are automatically generated by conda-lock.

Do you know what should be the right way to proceed?

File: src/sage/groups/braid.py [hunk 9]
```diff
@@ -2239,6 +2240,90 @@ def colored_jones_polynomial(self, N, variab=None, try_inverse=True):
...
+            sage: b.send_to_sc()
+            (s0*s1*s0*s2*s1, s0^2*s1*s2)
+        """
+        to_sc = send_to_sc(self)
+        B = self.parent()
+        return tuple([B._element_from_libbraiding(b) for b in to_sc])
+
+    def trajectory(self):
+        r"""
+        Return the braid's trajectory.
```
Comment (enriqueartal) [inline src/sage/groups/braid.py:2293]: I guess a little more explanation would help.

File: src/sage/groups/braid.py [hunk 10]
```diff
@@ -2239,6 +2240,90 @@ def colored_jones_polynomial(self, N, variab=None, try_inverse=True):
...
+             s0*s1*s2*s1^2,
+             s0*s1*s0*s2*s1]
+        """
+        traj = trajectory(self)
+        B = self.parent()
+        return [B._element_from_libbraiding(b) for b in traj]
+
+    def cyclic_slidings(self):
+        r"""
+        Return the braid's cyclic slidings.
```
Comment (enriqueartal) [inline src/sage/groups/braid.py:2311]: Same thing

**Review by enriqueartal — COMMENTED** [review]
I find it OK, and I agreed with @tscrim comments. I add a couple of comments about documentation. Basically it is fine for me.

**Comment by enriqueartal** [comment]
> Thanks for the quick review!
> 
> There is still a problem with the install of the new version of libbraiding that I don't know how to solve: in the CI environments the new version is not installed, and then the tests fail.
> 
> I guess it has something to do with the `environment-3*.yaml` files, which are automatically generated by conda-lock.
> 
> Do you know what should be the right way to proceed?

Is it a more general problem? I updated my system to Fedora 41, and `sirocco` did not work after a compilation because of python's version. And it could not be reinstalled because it was already there.

**Comment by miguelmarco** [comment]
I think I managed to force the CI systems to install the right version. Tests should pass now.

**Comment by enriqueartal** [comment]
Since the test pass, it seems OK to a positive review.

**Review by enriqueartal — APPROVED** [review]

**Pull request merged**